central/graphql/resolvers/plotted_node_vulnerabilities.go

```diff
@@ -24,16 +27,50 @@ type PlottedNodeVulnerabilitiesResolver struct {
 }
 
 func newPlottedNodeVulnerabilitiesResolver(ctx context.Context, root *Resolver, args RawQuery) (*PlottedNodeVulnerabilitiesResolver, error) {
-	query := withNodeCveTypeFiltering(args.String())
-	allCveIds, fixableCount, err := getPlottedVulnsIdsAndFixableCount(ctx, root, RawQuery{Query: &query})
+	if !features.PostgresDatastore.Enabled() {
+		q := withNodeCveTypeFiltering(args.String())
+		allCveIds, fixableCount, err := getPlottedVulnsIdsAndFixableCount(ctx, root, RawQuery{Query: &q})
+		if err != nil {
+			return nil, err
+		}
+
+		return &PlottedNodeVulnerabilitiesResolver{
+			root:    root,
+			all:     allCveIds,
+			fixable: fixableCount,
+		}, nil
+	}
+
+	query, err := getPlottedVulnsV1Query(args)
+	if err != nil {
+		return nil, err
+	}
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	allCves, err := vulnLoader.FromQuery(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	allCveIds := make([]string, 0, len(allCves))
+	for _, cve := range allCves {
+		allCveIds = append(allCveIds, cve.GetId())
+	}
+
+	ErrorOnQueryContainingField(query, search.Fixable, "Unexpected `Fixable` field in PlottedNodeVulnerabilities resolver")
```

Comment:
I'd say move this function to right after you create the query object and check for error on that. Otherwise there's a chance this doesn't get logged